doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,460 @@
+A quantum function with mid-circuit measurements can be executed via the
+`deferred measurement principle <https://en.wikipedia.org/wiki/Deferred_Measurement_Principle>`__.
+In PennyLane, this technique is available via ``mcm_method="deferred"`` or as the
+transform :func:`~.pennylane.defer_measurements`.
+
+The deferred measurement principle provides a powerful method to simulate
+mid-circuit measurements, conditional operations and measurement statistics
+in a differentiable and device-independent way. It adds an auxiliary qubit
+to the circuit for each mid-circuit measurement, leading to overheads of both
+memory and simulation time that scale exponentially with the number of measurements.
+
+.. code-block:: pycon
+
+    >>> deferred_qnode = qml.defer_measurements(my_qnode)
+    >>> pars = np.array([0.643, 0.246])
+    >>> deferred_qnode(*pars)
+    (tensor([0.90165331, 0.09834669], requires_grad=True),
+     tensor(0.09984972, requires_grad=True))
+
+The effect of deferring measurements becomes clear if we draw the ``QNode``
+before and after applying the transform:
+
+.. code-block:: pycon
+
+    >>> qml.draw(my_qnode)(*pars)
+    0: ──RY(0.64)─╭●───────RY(0.25)─┤  Probs
+    1: ───────────╰X──┤↗├──║────────┤
+                       ╚═══╩════════╡  <MCM>
+    >>> qml.draw(deferred_qnode)(*pars)
+    0: ──RY(0.64)─╭●────╭RY(0.25)─┤  Probs
+    1: ───────────╰X─╭●─│─────────┤
+    2: ──────────────╰X─╰●────────┤  <None>
+
+Mid-circuit measurements are deferred to the end of the circuit, and conditionally applied
+operations become (quantumly) controlled operations.
+
+.. note::
+
+    This method requires an additional qubit for each mid-circuit measurement, which limits
+    the number of measurements that can be used both on classical simulators and quantum hardware.
+
+    Postselection with deferred measurements is only supported on
+    :class:`~.pennylane.devices.DefaultQubit`.
+
+
+.. _one_shot_transform:
+
+Dynamic one-shot sampling
+*************************
+
+Devices that natively support mid-circuit measurements can evaluate dynamic circuits
+by executing them one shot at a time, sampling a dynamic execution path for each shot.
+
+In PennyLane, this technique is available via the QNode argument ``mcm_method="one-shot"``
+or as the transform :func:`~.pennylane.dynamic_one_shot`.
+As the name suggests, this transform only works for a :class:`~.pennylane.QNode` executing
+with finite shots and it requires the device to support mid-circuit measurements natively.
+
+The :func:`~.pennylane.dynamic_one_shot` transform is usually advantageous compared
+with the :func:`~.pennylane.defer_measurements` transform in the
+many-mid-circuit-measurements and few-shots limit. This is because, unlike the
+deferred measurement principle, the method does not need an additional wire for every
+mid-circuit measurement in the circuit.
+
+.. warning::
+
+    Dynamic circuits executed with shots should be differentiated with the finite difference method.
```

Comment:
Extremely optional but with the ongoing discussion with finite shots and finite diff, should some information about that be included here? This may not be the right document for that discussion though.

Reply:
I think we mention the fallback next to the support table, and that `h` should be set there. Otherwise I agree that it is not an MCM-specific topic, so I would not include it here.